app/src/components/tags/HorizontalScrollableTagViewer.vue

```diff
@@ -0,0 +1,95 @@
+<script setup lang="ts">
+import type { Tag } from "@/types";
+import PostTile from "@/components/posts/PostTile.vue";
+import { usePostStore } from "@/stores/post";
+import { storeToRefs } from "pinia";
+import { ArrowLeftCircleIcon, ArrowRightCircleIcon } from "@heroicons/vue/24/solid";
+import { ref, onMounted } from "vue";
+
+const postStore = usePostStore();
+
+const { postsByTag } = storeToRefs(postStore);
+
+type Props = {
+    tag: Tag;
+};
+const props = defineProps<Props>();
+
+const spinLeft = () => {
+    if (scrollElement.value) scrollElement.value.scrollLeft -= 100;
+};
+const spinRight = () => {
+    if (scrollElement.value) scrollElement.value.scrollLeft += 100;
+};
+
+const showLeftSpin = ref(false);
+const showRightSpin = ref(false);
+
+const scrollElement = ref<HTMLElement | null>(null);
+
+const setSpinBtnVisibility = () => {
+    if (scrollElement.value === null) return;
+
+    if (scrollElement.value.scrollWidth <= scrollElement.value.clientWidth) {
+        showLeftSpin.value = false;
+        showRightSpin.value = false;
+        return;
+    }
+
+    // Check if the browser is in portrait mode, indicating that a mobile device is being used
+    if (window.matchMedia("(orientation: portrait)").matches) {
+        showLeftSpin.value = false;
+        showRightSpin.value = false;
+        return;
+    }
+
+    const scrollRight =
+        scrollElement.value.scrollWidth -
+        scrollElement.value.clientWidth -
+        scrollElement.value.scrollLeft;
+    const scrollLeft = scrollElement.value.scrollLeft;
+    console.log(scrollRight);
+    if (scrollLeft > 20) showLeftSpin.value = true;
+    if (scrollLeft < 5) showLeftSpin.value = false;
+    if (scrollRight > 20) showRightSpin.value = true;
+    if (scrollRight < 5) showRightSpin.value = false;
+};
+
+onMounted(() => {
+    // wait for child controls in the scroll element to be rendered
+    setTimeout(() => {
+        setSpinBtnVisibility();
+    }, 100);
+});
```

Comment:
Instead, you could also add a one-time `watch` on `scrollElement`, which could be a bit more reliable.